src/main/java/com/main/server/controllers/ProductController.java

```diff
@@ -25,22 +26,26 @@ public class ProductController {
 
     private ProductService productService;
 
-    @PostMapping("/product")
+    @PreAuthorize("hasPermission(#product.organization().name(), 'edit')")
+    @PostMapping("/products")
     public ResponseEntity<Product> saveProduct(@RequestBody Product product, @PathVariable @NotNull Long orgId) {
         return ResponseEntity.ok(productService.saveProduct(product, orgId));
     }
 
-    @GetMapping("/product/{id}")
+    @GetMapping("/products/{id}")
     public ResponseEntity<Product> getProductById(@PathVariable @NotNull Long id) {
         return ResponseEntity.ok(productService.getProductById(id));
     }
 
-    @PutMapping("/product")
-    public ResponseEntity<Product> updateProduct(@RequestBody Product product, @PathVariable @NotNull Long orgId) {
-        return ResponseEntity.ok(productService.updateProduct(product, orgId));
+    @PreAuthorize("hasPermission(#product.organization().name(), 'edit')")

```

Comment:
we need to change logic in permissionEvaluator, we need to pass only organizationId to check permission